plugin/src/main/kotlin/com/varabyte/kobweb/intellij/startup/KobwebPostStartupProjectActivity.kt

```diff
@@ -1,25 +1,106 @@
 package com.varabyte.kobweb.intellij.startup
 
+import com.intellij.notification.NotificationType
+import com.intellij.openapi.actionSystem.*
+import com.intellij.openapi.actionSystem.ex.ActionUtil
+import com.intellij.openapi.actionSystem.ex.AnActionListener
 import com.intellij.openapi.components.service
+import com.intellij.openapi.externalSystem.action.RefreshAllExternalProjectsAction
 import com.intellij.openapi.externalSystem.service.project.manage.ProjectDataImportListener
 import com.intellij.openapi.project.Project
+import com.intellij.openapi.project.modules
+import com.intellij.openapi.project.rootManager
+import com.intellij.openapi.roots.LibraryOrderEntry
+import com.intellij.openapi.roots.ModuleOrderEntry
 import com.intellij.openapi.startup.ProjectActivity
+import com.varabyte.kobweb.intellij.notification.KobwebNotificationHandle
+import com.varabyte.kobweb.intellij.notification.KobwebNotifier
 import com.varabyte.kobweb.intellij.services.project.KobwebProjectCacheService
+import com.varabyte.kobweb.intellij.util.kobweb.KobwebPluginState
+import com.varabyte.kobweb.intellij.util.kobweb.kobwebPluginState
+
+/**
+ * A simple heuristic for checking if this project has code in it somewhere that depends on Kobweb.
+ *
+ * This is useful as users may install the Kobweb plugin for one or two of their projects, but we should stay out of the
+ * way in every other kind of project.
+ */
+private fun Project.hasAnyKobwebDependency(): Boolean {
+    return this.modules.asSequence()
+        .flatMap { module -> module.rootManager.orderEntries.asSequence() }
+        .any { orderEntry ->
+            when (orderEntry) {
+                // Most projects will indicate a dependency on Kobweb via library coordinates, e.g. `com.varabyte.kobweb:core`
+                is LibraryOrderEntry -> orderEntry.libraryName.orEmpty().substringBefore(':') == "com.varabyte.kobweb"
+                // Very rare, but if a project depends on Kobweb source directly, that counts. This is essentially for
+                // the `kobweb/playground` project which devs use to test latest Kobweb on.
+                is ModuleOrderEntry -> orderEntry.moduleName.substringBefore('.') == "kobweb"
+                else -> false
+            }
+        }
+}
 
 /**
  * Actions to perform after the project has been loaded.
  */
 class KobwebPostStartupProjectActivity : ProjectActivity {
-    private class ImportListener(private val project: Project) : ProjectDataImportListener {
+    private class ImportListener(
```

Comment:
Could we not implement the start/finish hooks in the gradle resolver?  Now there are two separate places where we implement project imports

Reply:
This is definitely more art than science :)

Note that this approach lets me dismiss the Kobweb popup, which I don't think would be easy to do in the Gradle resolver.

But beyond that, I think there's value doing this in code that is decoupled not only from the Gradle resolver but from Gradle in general.

If someone happens to have the Kobweb plugin installed and then opens up a project that doesn't use Gradle at all, this expresses the idea that this code is still relevant in that case.

> Now there are two separate places where we implement project imports

I think about this differently. Currently, the GradleProjectResolver class is focused on one thing only, which is resolving Gradle projects and setting metadata if found. This import listener is focused on updating the Kobweb plugin with latest state. Both have their own purpose.

The GradleProjectResolver "implements project imports", while the import listener is more of a side-effect updater IMO.

Reply:
Fair. I also dont mind it a lot